MekHQ/src/mekhq/campaign/Warehouse.java

```diff
@@ -263,13 +271,52 @@ private Part mergePartWithExisting(Part part) {
      * @return The matching spare part or null if none were found.
      */
     public @Nullable Part checkForExistingSparePart(Part part) {
-        Objects.requireNonNull(part);
+        if (part == null) {
+            logger.error("checkForExistingSparePart(Part): Part is null");
```

Comment:
This is not an error, it's a debug, Maybe an `info`, but it will show up in the log as an error and it will mean nothing, it won't have any kind of information, wont show an ID, nothing, so its an error log that will exist but no one will be able to do anything to act upon this error being logged.